### PR TITLE
BATIAI-2213-Migrate the rest of the IRSA modules into batcave-tf-utility-belt-irsa

### PR DESCRIPTION
**Pull request by skumbala40**
### Submitter Checklist
-  [ ] Is there an impact on Auditing and Logging procedures or capabilities?
-  [ ] Is there an impact on Authentication procedures or capabilities?
-  [ ] Is there an impact on Authorization procedures or capabilities?
-  [ ] Is there an impact on Communication Security procedures or capabilities?
-  [ ] Is there an impact on Cryptography procedures or capabilities?
-  [ ] Is there an impact on Sensitive Data procedures or capabilities?
-  [ ] Is there an impact on any other security-related procedures or capabilities?
-  [x] No security impacts identified.

Security Risks Identified - For any applicable items on the "Submitter Checklist," describe the impact of the change and any implemented mitigations.

### CODE REVIEW

**Comment by robo-gotham** [comment]
Snyk Scanning for Commit: abea0f9ec8109f7026cbaa0b29c32146c25a899a

Snyk Infrastructure as Code

- Snyk testing Infrastructure as Code configuration issues.
✔ Test completed.

Issues
  No vulnerable paths were found!

-------------------------------------------------------

</details>

Test Summary

  Organization: batcave-ispg
  Project name: CMS-Enterprise/batcave-tf-utility-belt-irsa

✔ Files without issues: 3
✗ Files with issues: 0
  Ignored issues: 0
  Total issues: 0 [ 0 critical, 0 high, 0 medium, 0 low ]

-------------------------------------------------------

Report Complete

  Your test results are available at: https://snyk.io/org/batcave-ispg/projects
  under the name: CMS-Enterprise/batcave-tf-utility-belt-irsa

File: variable.tf [hunk 1]
```diff
@@ -51,3 +51,87 @@ variable "velero_service_accounts" {
   type    = list(string)
   default = ["velero:velero-velero-server"]
 }
+
+variable "argocd_sops_arn" {
+  type        = string
+  description = "sops for argocd"
```
Comment (bushong1) [inline variable.tf:57]:
```suggestion
  description = "sops kms key arn for argocd"
```

File: variable.tf [hunk 2]
```diff
@@ -51,3 +51,87 @@ variable "velero_service_accounts" {
...
+}
+variable "container-insights_service_accounts" {
+  type    = list(string)
+  default = ["amazon-cloudwatch:aws-cloudwatch-metrics"]
+}
+
+variable "gitlab_bucket_arns" {
+  type        = list(string)
+  description = "buckets for gitlab"
+}
```
Comment (bushong1) [inline variable.tf:111]: This should not be required as not every environment will use gitlab.  default to empty list, configure the resources to only create if the length is > 0

File: variable.tf [hunk 3]
```diff
@@ -51,3 +51,87 @@ variable "velero_service_accounts" {
...
+  type = list(string)
+  default = [
+    "gitlab:gitlab-minio",
+    "gitlab:gitlab-webservice",
+    "gitlab:gitlab-toolbox",
+    "gitlab:gitlab-sidekiq"
+  ]
+}
+
+variable "gitlab-runner_bucket_arns" {
```
Comment (bushong1) [inline variable.tf:126]: resource names should only use underscores
```suggestion
variable "gitlab_runner_bucket_arns" {
```

File: variable.tf [hunk 4]
```diff
@@ -51,3 +51,87 @@ variable "velero_service_accounts" {
...
+  type        = list(string)
+  description = "buckets for gitlab-runner"
+}
+variable "gitlab-runner_app_name" {
```
Comment (bushong1) [inline variable.tf:130]:
```suggestion
variable "gitlab_runner_app_name" {
```

File: variable.tf [hunk 5]
```diff
@@ -51,3 +51,87 @@ variable "velero_service_accounts" {
...
+  type    = string
+  default = "GitlabRunner"
+}
+variable "gitlab-runner_service_accounts" {
```
Comment (bushong1) [inline variable.tf:134]:
```suggestion
variable "gitlab_runner_service_accounts" {
```

File: variable.tf (hunk 3 above)
Comment (bushong1) [reply variable.tf:126]: Same comment applies about making this variable optional; default to empty list, conditionally create resources

**Review by bushong1 — CHANGES_REQUESTED** [review]

File: variable.tf
[the remark this answers is not in the document]
Comment (skumbala40) [reply variable.tf:1]: updated file name.

File: variable.tf (hunk 1 above)
Comment (skumbala40) [reply variable.tf]: updated as suggested

File: variable.tf (hunk 3 above)
Comment (skumbala40) [reply variable.tf]: updated as suggested

File: variable.tf (hunk 4 above)
Comment (skumbala40) [reply variable.tf]: updated as suggested

File: variable.tf (hunk 5 above)
Comment (skumbala40) [reply variable.tf]: updated as suggested

**Review by j-mowery — APPROVED** [review]

**Review by bushong1 — APPROVED** [review]

**Pull request merged**